db/user.js

```diff
@@ -10,8 +11,22 @@ userSchema.pre('save', function (next) {
   const password = this.password;
   hash(password).then(hash => {
     this.password = hash;
-    next();
+  next();
   });
   });
 
+userSchema.pre('update', function (next) {
+  const password = this.password;
+  hash(password).then(hash => {
+    this.password = hash;
+  next();
+  });
+  });
+
+Object.assign(userSchema.methods, {
+  validPassword(password, cb) {
+    bcrypt.compare(password, this.password, cb);
+  }
+});
```

Comment:
Two things:
1. This method should be called either `validatePassword` or `isValidPassword` since it returns if password is valid. (I prefer the first option but take a pick or go with something along those lines)
2. You are not returning anything which means that consumer of this method will still get no info about weather password is valid or not. 🐐